pkg/controllers/namespace/namespace_controller.go

```diff
@@ -73,22 +79,26 @@ func (r *NamespaceReconciler) createVPCCR(ctx *context.Context, obj client.Objec
 		return nil, errors.New(message)
 	}
 
-	// create vpc cr with existing vpc network config
-	vpcCR := BuildVPCCR(ns, ncName, vpcName)
-	err := r.Client.Create(*ctx, vpcCR)
+	// create networkInfo cr with existing vpc network config
+	var err error
+	if networkInfoCR == nil {
+		log.V(2).Info("building networkInfo", "ns", ns)
+		networkInfoCR = &v1alpha1.NetworkInfo{}
+		networkInfoCR.Name = NETWORKINFOPREFIX + uuid.New().String()
+		networkInfoCR.Namespace = ns
+		networkInfoCR.VPCs = []v1alpha1.VPCState{}
+		err = r.Client.Create(*ctx, networkInfoCR)
+	} else {
+		err = r.Client.Update(*ctx, networkInfoCR)
+	}
```

Comment:
I do not get the point here, Why need call r.Client.Update()?
can use BuildNetworkInfoCR() func here.

Reply:
> can use BuildNetworkInfoCR() func here.

oh, I removed the file builder.go since it's too short

> Why need call r.Client.Update()?
When vpcnetworkconfig changs, such as  ExternalIPv4Blocks and PrivateIPv4CIDRs, then should update it.

Reply:
OK, move vpcnetworkconfig_handler to networkinfo, it should avoid this problem in this way.